src/software/backend/simulation/simulator_robot.cpp

```diff
@@ -0,0 +1,185 @@
+#include "software/backend/simulation/simulator_robot.h"
+
+#include "app/world/chicker.h"
+#include "app/world/dribbler.h"
+#include "app/world/wheel.h"
+
+std::optional<unsigned int> SimulatorRobot::robot_id                    = std::nullopt;
+std::vector<std::weak_ptr<PhysicsRobot>> SimulatorRobot::physics_robots = {};
+
+void SimulatorRobot::setRobotId(unsigned int id)
+{
+    robot_id = std::make_optional<unsigned int>(id);
+}
+
+void SimulatorRobot::setPhysicsRobots(
+    const std::vector<std::weak_ptr<PhysicsRobot>>& robots)
+{
+    physics_robots = robots;
+}
+
+FirmwareRobot_t* SimulatorRobot::createFirmwareRobot()
+{
+    Chicker_t* chicker = app_chicker_create(
+        &(SimulatorRobot::kick), &(SimulatorRobot::chip),
+        &(SimulatorRobot::enableAutokick), &(SimulatorRobot::enableAutochip),
+        &(SimulatorRobot::disableAutokick), &(SimulatorRobot::disableAutochip));
+
+    Dribbler_t* dribbler =
+        app_dribbler_create(&(SimulatorRobot::setDribblerSpeed),
+                            &(SimulatorRobot::getDribblerTemperatureDegC));
+
+    Wheel_t* front_left_wheel =
+        app_wheel_create(&(SimulatorRobot::applyWheelForceFrontLeft));
+    Wheel_t* front_right_wheel =
+        app_wheel_create(&(SimulatorRobot::applyWheelForceFrontRight));
+    Wheel_t* back_left_wheel =
+        app_wheel_create(&(SimulatorRobot::applyWheelForceBackLeft));
+    Wheel_t* back_right_wheel =
+        app_wheel_create(&(SimulatorRobot::applyWheelForceBackRight));
+
+    FirmwareRobot_t* firmware_robot =
+        app_firmware_robot_create(chicker, dribbler, &(SimulatorRobot::getPositionX),
+                                  &(SimulatorRobot::getPositionY), front_right_wheel,
+                                  front_left_wheel, back_right_wheel, back_left_wheel);
```

Comment:
Just as a note, currently we rely on the user to destruct each object they create, ie. `FirmwareRobot_t` does not take ownership of the objects you pass to it. We can look at changing this, but let's just make sure we're making an explicit choice.

Reply:
Ah right. IIRC you were going to change that in a future PR, so that the object takes ownership of anything injected to it? That would make things here a lot easier. I can make an explicit note / ticket to change it or look into it again in the future. Should be ok to leak here for the very short term.

Reply:
made #1128